### PR TITLE
Update system software documentation

### PR DESCRIPTION
**Pull request by UniversalSuperBox**
This PR changes the "App development" chapter to "Development" and adds a new System software development section underneath it. It also updates the system software development documentation.

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -49,10 +49,11 @@ You may view this documentation in the following languages:
 
 .. toctree::
    :maxdepth: 1
-   :caption: App development
+   :caption: Development
    :name: sec-appdev
```
Comment (UniversalSuperBox) [inline index.rst]: Rename this section

File: systemdev/testing-locally.rst [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+We only recommend developing packages using a device with Ubuntu Touch installed from the devel channel. This ensures that you are testing your changes against the most current state of the Ubuntu Touch code.
+
+.. warning::
+
+    Installing packages has a risk of damaging the software on your device, rendering it unusable. If this happens, you can :doc:`reinstall Ubuntu Touch </userguide/install>`.
+
+Cross-building with crossbuilder
+--------------------------------
+
+This is the recommended way to develop non-trivial changes, and it's suitable for all devices since the build happens on your desktop PC (we will call it "host" from now on) and not on the target device.
```
Comment (UniversalSuperBox) [inline systemdev/testing-locally.rst]: Run-on sentence. Reword to avoid confusion.

File: systemdev/testing-locally.rst [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    cd ~
+    git clone https://github.com/ubports/crossbuilder.git
+
+It's a shell script, so you don't need to build it. Instead, you will need to add its directory to your ``PATH`` environment variable, so that you can execute it from any directory::
+
+    echo "export PATH=$HOME/crossbuilder:$PATH" >> ~/.bashrc
+    # and add it to your own session:
+    export PATH="$HOME/crossbuilder:$PATH"
+
+Then, you need to setup LXD; luckily, crossbuilder has a command which does everything for you; you just need to carefully follow its instructions::
```
Comment (UniversalSuperBox) [inline systemdev/testing-locally.rst]: Too much punctuation. Reword sentence to simplify its structure.

**Comment by bhdouglass** [comment]
@UniversalSuperBox were you going to make the changes you suggested or should we merge and update them later?

**Comment by UniversalSuperBox** [comment]
There, all fixed. Ready for review @bhdouglass!

**Pull request merged**